### PR TITLE
418288 Dashboard error handling

### PR DESCRIPTION
**Pull request by johnbenjamin-defra**

### CODE REVIEW

**Review by udaydefra — DISMISSED** [review]

**Review by trilokkhandelwal — APPROVED** [review]
Approved

**Review by trilokkhandelwal — DISMISSED** [review]
Approve

File: src/EPR.Calculator.Frontend/Controllers/DashboardController.cs [hunk 1]
```diff
@@ -27,35 +27,43 @@ public DashboardController(IConfiguration configuration, IHttpClientFactory clie
...
+
+                    return this.View(ViewNames.DashboardIndex, dashboardRunData);
+                }
+
+                if (response.Result.StatusCode == HttpStatusCode.NotFound)
+                {
+                    return this.View();
+                }
+
+                return RedirectToAction("Index", "StandardError");
```
Comment (udaydefra) [inline src/EPR.Calculator.Frontend/Controllers/DashboardController.cs:61]: Please use constants for view names

File: src/EPR.Calculator.Frontend/Controllers/DashboardController.cs [hunk 2]
```diff
@@ -27,35 +27,43 @@ public DashboardController(IConfiguration configuration, IHttpClientFactory clie
...
+                var request = new HttpRequestMessage(HttpMethod.Post, new Uri(dashboardcalculatorrunApi));
 
-            if (response.Result.IsSuccessStatusCode)
-            {
-                var dashboardRunData = this.GetCalulationRunsData(JsonConvert.DeserializeObject<List<CalculationRun>>(response.Result.Content.ReadAsStringAsync().Result));
+                var runParms = new CalculatorRunParamsDto
+                {
+                    FinancialYear = year,
+                };
+                var content = new StringContent(JsonConvert.SerializeObject(runParms), System.Text.Encoding.UTF8, "application/json");
```
Comment (udaydefra) [inline src/EPR.Calculator.Frontend/Controllers/DashboardController.cs:42]: Please create a constants for application/json

File: src/EPR.Calculator.Frontend/Controllers/DashboardController.cs [hunk 3]
```diff
@@ -27,35 +27,43 @@ public DashboardController(IConfiguration configuration, IHttpClientFactory clie
...
+                    return this.View();
+                }
+
+                return RedirectToAction("Index", "StandardError");
+            }
+            catch (Exception ex)
             {
-                ViewBag.IsDataAvailable = false;
-                return this.View();
+                return RedirectToAction("Index", "StandardError");
```
Comment (udaydefra) [inline src/EPR.Calculator.Frontend/Controllers/DashboardController.cs:65]: Please use constants for view names

File: src/EPR.Calculator.Frontend/Controllers/DashboardController.cs [hunk 4]
```diff
@@ -27,35 +27,43 @@ public DashboardController(IConfiguration configuration, IHttpClientFactory clie
...
-            var runParms = new CalculatorRunParamsDto
+            try
             {
-                FinancialYear = year,
-            };
-            var content = new StringContent(JsonConvert.SerializeObject(runParms), System.Text.Encoding.UTF8, "application/json");
-            request.Content = content;
-            var response = client.SendAsync(request);
-            response.Wait();
+                var dashboardcalculatorrunApi = this.configuration.GetSection("DashboardCalculatorRun").GetSection("DashboardCalculatorRunApi").Value;
```
Comment (udaydefra) [inline src/EPR.Calculator.Frontend/Controllers/DashboardController.cs:32]: Again constants for DashboardCalculatorRun and DashboardCalculatorRunApi

File: src/EPR.Calculator.Frontend/Controllers/DashboardController.cs (hunk 1 above)
Comment (johnbenjamin-defra) [reply src/EPR.Calculator.Frontend/Controllers/DashboardController.cs]: I have updated the code.

File: src/EPR.Calculator.Frontend/Controllers/DashboardController.cs (hunk 2 above)
Comment (johnbenjamin-defra) [reply src/EPR.Calculator.Frontend/Controllers/DashboardController.cs]: I have updated the code. Thanks

File: src/EPR.Calculator.Frontend/Controllers/DashboardController.cs (hunk 3 above)
Comment (johnbenjamin-defra) [reply src/EPR.Calculator.Frontend/Controllers/DashboardController.cs]: I have updated the code.

File: src/EPR.Calculator.Frontend/Controllers/DashboardController.cs (hunk 4 above)
Comment (johnbenjamin-defra) [reply src/EPR.Calculator.Frontend/Controllers/DashboardController.cs:33]: This is not possible as getting the below error:

_The expression being assigned to must be constant._

Comment (udaydefra) [reply src/EPR.Calculator.Frontend/Controllers/DashboardController.cs:33]: No worries, Will see later

**Review by udaydefra — APPROVED** [review]

**Pull request merged**